src/store/dirinfo.ts

```diff
@@ -39,14 +39,12 @@ class DirInfoStore {
   }
 
   public get xlDatasets(): string[] {
-    const xlDatasets: string[] = []
-
-    if (this.dirInfoData?.dsDict) {
-      Object.values(this.dirInfoData?.dsDict).forEach(
-        item => item.kind === 'xl' && xlDatasets.push(item.name),
-      )
-    }
-    return xlDatasets
+    return this.dirInfoData?.dsDict
+      ? Object.values(this.dirInfoData?.dsDict)
+          .filter(Boolean)
+          .filter(item => item.kind === 'xl')
```

Comment:
If you use the second filter why do you need the first one with Boolean?
Maybe it's enough to check `item?.kind === 'xl'`

Reply:
Fixed, but item!.name should be, because otherwise will be compile error